src/main/java/seedu/ecardnomics/game/GameEngine.java

```diff
@@ -0,0 +1,139 @@
+package seedu.ecardnomics.game;
+
+import seedu.ecardnomics.Ui;
+import seedu.ecardnomics.command.Command;
+import seedu.ecardnomics.command.ExitCommand;
+import seedu.ecardnomics.command.game.DoneGameCommand;
+import seedu.ecardnomics.command.game.GameResponseCommand;
+import seedu.ecardnomics.deck.FlashCard;
+import seedu.ecardnomics.parser.GameParser;
+
+public class GameEngine {
+    GameStorage storage;
+    GameParser gameParser;
+
+    GameEngine(GameStorage storage) {
+        gameParser = new GameParser(storage.originalDeck);
+        this.storage = storage;
+    }
+
+    Command runGameLoop() {
+        Command command;
+
+        do {
+            FlashCard flashCard = poseQuestion();
+            command = getAttempt();
```

Comment:
Idea for UG FAQ: What if my flashcard answer contains a single word "done" or "exit"?
Possible answer: Escape the command using \exit or "exit".

Reply:
^ sounds good! I think in checkAttempt() method excluding {Punc} probably already catches "\", if not then we can just add that to the regex to ignore. Great idea!

Reply:
^ Just checked, can confirm {Punct} regex includes "\". Will add this to FAQ portion of UG!